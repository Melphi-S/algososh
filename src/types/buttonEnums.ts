export enum Direction {
  Ascending = "ascending",
  Descending = "descending",
}

export enum Action {
  Add = "add",
  Delete = "delete",
}
