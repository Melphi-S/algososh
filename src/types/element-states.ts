export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
  Swaping = "swaping",
}
