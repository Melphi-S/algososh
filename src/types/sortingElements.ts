import {ElementStates} from "./element-states";

export type TCharElement = {
    char: string;
    id: number;
    state: ElementStates;
}