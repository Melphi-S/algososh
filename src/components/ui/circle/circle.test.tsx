import {Circle} from "./circle";
import {cleanup, render, screen, within} from "@testing-library/react";
import {ElementStates} from "../../../types/element-states";
import React from "react";

describe('Circle', () => {
    beforeEach(cleanup);
    it('should render without letters', function () {
        render(<Circle/>);
        const circle = screen.getByTestId("circle");
        expect(circle).toHaveTextContent('');
        expect(circle).toMatchSnapshot();
    });

    it('should render with letters', function () {
        render(<Circle letter='test'/>);
        const circle = screen.getByTestId("circle");
        expect(circle).toHaveTextContent('test');
        expect(circle).toMatchSnapshot();
    });

    it('should render with head as text', function () {
        render(<Circle head='head'/>);
        const circle = screen.getByTestId("circle");
        const head = screen.getByTestId("head");
        expect(head).toHaveTextContent('head');
        expect(head).toHaveClass('string');
        expect(circle).toMatchSnapshot();
    });

    it('should render with head as react-element', function () {
        render(<Circle head={<Circle letter='test' isSmall />}/>);
        const head = screen.getAllByTestId("head");
        const circle = within(head[0]).queryByTestId('circle');
        expect(circle).toBeInTheDocument();
        expect(head[0]).toHaveClass('element');
        expect(head).toMatchSnapshot();
    });

    it('should render with tail as text', function () {
        render(<Circle tail='tail'/>);
        const circle = screen.getByTestId("circle");
        const tail = screen.getByTestId("tail");
        expect(tail).toHaveTextContent('tail');
        expect(tail).toHaveClass('string');
        expect(circle).toMatchSnapshot();
    });

    it('should render with tail as react-element', function () {
        render(<Circle tail={<Circle letter='test' isSmall />}/>);
        const tail = screen.getAllByTestId("tail");
        const circle = within(tail[0]).queryByTestId('circle');
        expect(circle).toBeInTheDocument();
        expect(tail[0]).toHaveClass('element');
        expect(tail).toMatchSnapshot();
    });

    it('should render with index', function () {
        render(<Circle index={1}/>);
        const circle = screen.getByTestId("circle");
        const index = within(circle).getByTestId('index');
        expect(index).toHaveTextContent('1');
        expect(circle).toMatchSnapshot();
    });

    it('should render with small-prop', function () {
        render(<Circle isSmall/>);
        const circle = screen.getByTestId("circle");
        const circleCore = within(circle).getByTestId('circle-core');
        expect(circleCore).toHaveClass('small');
        expect(circle).toMatchSnapshot();
    });

    it('should render with default state', function () {
        render(<Circle state={ElementStates.Default}/>);
        const circle = screen.getByTestId("circle");
        const circleCore = within(circle).getByTestId('circle-core');
        expect(circleCore).toHaveClass('default');
        expect(circle).toMatchSnapshot();
    });

    it('should render with changing state', function () {
        render(<Circle state={ElementStates.Changing}/>);
        const circle = screen.getByTestId("circle");
        const circleCore = within(circle).getByTestId('circle-core');
        expect(circleCore).toHaveClass('changing');
        expect(circle).toMatchSnapshot();
    });

    it('should render with modified state', function () {
        render(<Circle state={ElementStates.Modified}/>);
        const circle = screen.getByTestId("circle");
        const circleCore = within(circle).getByTestId('circle-core');
        expect(circleCore).toHaveClass('modified');
        expect(circle).toMatchSnapshot();
    });
})