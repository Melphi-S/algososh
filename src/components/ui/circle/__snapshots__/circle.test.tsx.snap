// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle should render with changing state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   changing"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with default state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with head as react-element 1`] = `
Array [
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        class="circle  small default"
        data-testid="circle-core"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          test
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>,
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />,
]
`;

exports[`Circle should render with head as text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    head
  </div>
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with index 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with letters 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      test
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with modified state 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   modified"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with small-prop 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle  small default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Circle should render with tail as react-element 1`] = `
Array [
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      class="content "
      data-testid="circle"
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        class="circle  small default"
        data-testid="circle-core"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          test
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
        data-testid="index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>,
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />,
]
`;

exports[`Circle should render with tail as text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Circle should render without letters 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    class="circle   default"
    data-testid="circle-core"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
